### ISSUE TITLE
Cannot define class methods using concerning

### ISSUE THREAD

**Issue by johnnaegle** [basecamp/concerning#1]
I've got this:

``` ruby
class User < ActiveRecord::Base
  attr_accessible :name

  concerning :Programmers do
    module ClassMethods 
      def programmer?
        true
      end
    end
  end

  module Managers 
    extend ActiveSupport::Concern

    module ClassMethods
      def manager?
        true
      end
    end
  end

  include Managers
end
```

And I would expect this to work:

```
User.manager?
User.programmer?
```

But the second raises `NoMethodError: undefined method `programmer?' for #<Class:0x007f9641beafd0>`

How can I define class methods with concerning?

**Comment by johnnaegle** [comment]
If I write it like this, it does work as expected, but it isn't nearly as clean:

``` ruby
  concerning :Programmers do
    included do
      extend User::ProgrammersClassMethods
    end

    module ProgrammersClassMethods 
      def programmer?
        true
      end
    end
  end
```

**Comment by jeremy** [comment]
Yes - can't do classmethods modules within the block. Plan is to introduce
a DSL for these instead, like `class_methods do ...`

**Comment by javan** [comment]
Another workaround: 

``` ruby
concerning :Programmers do
  included do
    def self.programmer?
      true
    end
  end
end
```

**Comment by robmathews** [comment]
@javan - that works

**Comment by barelyknown** [comment]
Just noticed this issue after I wrote this PR https://github.com/37signals/concerning/pull/2. It looks like my solution is consistent with @jeremy's https://github.com/37signals/concerning/issues/1#issuecomment-31141860.

**Comment by jeremy** [comment]
Merged #2 - check out `class_methods do ...` for declaring class methods. Thanks @barelyknown!
